tiledb/sm/c_api/tiledb.cc

```diff
@@ -7142,6 +7142,22 @@ int32_t tiledb_fragment_info_load_with_key(
   return TILEDB_OK;
 }
 
+int32_t tiledb_fragment_info_get_fragment_name(
```

Comment:
The function implementation here LGTM, however this function block needs to be moved inside of the `tiledb::common::detail` namespace.

Instead, here we need to have a function wrapper implementation returning the result of calling `api_entry<detail::...get_fragment_name(ctx, fragment_info, fid, name)` -- see similar entries in this section for examples.